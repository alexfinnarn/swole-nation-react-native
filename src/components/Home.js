import React, {useState} from 'react';
import {Button, Picker, Text, View, TouchableOpacity} from "react-native";
import {styles, home} from './Styles';
import SessionTeaserProvider from "./sessions/SessionTeaserProvider";

export default function Home({navigation, workouts, handle, thing}) {
  const [nextWorkoutId, setNextWorkoutId] = useState(workouts[0].id);

  return (
    <View style={[styles.container, {flex: 1, flexDirection: 'column', justifyContent: 'space-evenly'}]}>
      <View style={home.sectionContainer}>
        <View style={home.sectionLeft}>
          <Text style={home.sectionHeaderText}>Sessions</Text>
        </View>
        <View style={{flex: 1, flexDirection: 'column'}}>
          <TouchableOpacity
            style={[{backgroundColor: '#21897E'}, home.actionButton]}
            onPress={() => navigation.navigate('WorkoutsList')}>
            <View style={{flex: 1, justifyContent: 'center'}}>
              <Text style={home.actionButtonText}>Edit</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
      <View style={home.sectionContainer}>
        <View style={home.sectionLeft}>
          <Text style={home.sectionHeaderText}>Next Workout</Text>
          <View style={{flex: 2}}>
            <Picker
              selectedValue={nextWorkoutId}
              style={{height: 70, width: 140}}
              onValueChange={(value) => setNextWorkoutId(value)}>
              {workouts.map((workout) => <Picker.Item key={workout.id} label={workout.name} value={workout.id}/>)}
            </Picker>
          </View>
        </View>
        <View style={{flex: 1, flexDirection: 'column'}}>
          <TouchableOpacity
            style={[{backgroundColor: '#21897E'}, home.actionButton]}
            onPress={() => {
              handle.nextWorkoutInteraction(nextWorkoutId);
              navigation.navigate('Workout');
            }}>
            <View style={{flex: 1, justifyContent: 'center'}}>
              <Text style={home.actionButtonText}>Go</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
      <View style={home.sectionContainer}>
        <View style={home.sectionLeft}>
          <Text style={home.sectionHeaderText}>Workouts</Text>
        </View>
        <View style={{flex: 1, flexDirection: 'column'}}>
          <TouchableOpacity
            style={[{backgroundColor: '#21897E'}, home.actionButton]}
            onPress={() => navigation.navigate('WorkoutsList')}>
            <View style={{flex: 1, justifyContent: 'center'}}>
              <Text style={home.actionButtonText}>Edit</Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
      <View style={home.sectionContainer}>
        <View style={home.sectionLeft}>
          <Text style={home.sectionHeaderText}>Exercises</Text>
        </View>
        <TouchableOpacity
          style={[{backgroundColor: '#21897E'}, home.actionButton]}
          title="Edit"
          onPress={() => navigation.navigate('WorkoutsList')}>
          <View style={{flex: 1, justifyContent: 'center'}}>
            <Text style={home.actionButtonText}>Edit</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
}


