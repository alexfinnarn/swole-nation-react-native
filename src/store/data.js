import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
