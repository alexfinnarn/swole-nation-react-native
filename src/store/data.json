{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
