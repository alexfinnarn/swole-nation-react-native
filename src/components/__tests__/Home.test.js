import React from 'react';
import { render } from 'react-native-testing-library';
import data from '../../store/data'
import Home from '../Home';

const workouts = Object.keys(data.workouts).map((key) => data.workouts[key]);
const navigation = jest.fn();
const handle = jest.fn();

describe('<Home />', () => {
  it('First workout key selected in Picker', () => {
    const {getByTestId} = render(<Home workouts={workouts} navigation={navigation} handle={handle}/>);
    const element = getByTestId('workout-picker');
    expect(element.props.selectedValue).toBe('DQkECwYLCQQ');
  });
});
