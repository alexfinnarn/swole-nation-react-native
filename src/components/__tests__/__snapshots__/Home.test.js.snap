// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F5FCFF",
        "flex": 1,
        "flexDirection": "column",
        "paddingTop": 5,
      },
    ]
  }
  testID="home-root"
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginTop": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
          }
        }
      >
        Sessions
      </Text>
      <View
        style={
          Object {
            "flex": 3,
            "flexDirection": "column",
            "justifyContent": "space-evenly",
          }
        }
        testID="session-teaser-root"
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <Text
            testID="session-name"
          >
            Thu Sep 26 12:19:14 2019
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 2,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "column",
              }
            }
          >
            <Text>
              Workout:
            </Text>
            <Text
              testID="workout-name"
            >
              Texas Method A
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              }
            }
          >
            <Text>
              Completed:
            </Text>
            <Text
              testID="completion-percentage"
            >
              78%
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "column",
              }
            }
          >
            <Text>
              Duration:
            </Text>
            <Text
              testID="session-duration"
            >
              15
              s
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            List
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginTop": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
          }
        }
      >
        Next Workout
      </Text>
      <View
        style={
          Object {
            "flex": 2,
          }
        }
      >
        <View
          style={
            Object {
              "height": 70,
              "width": 160,
            }
          }
        >
          <RCTPicker
            items={
              Array [
                Object {
                  "label": "Texas Method A",
                  "textColor": undefined,
                  "value": "DQkECwYLCQQ",
                },
                Object {
                  "label": "Texas Method B",
                  "textColor": undefined,
                  "value": "DAwJBQEMAA0",
                },
              ]
            }
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={1}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
            testID="workout-picker"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Go
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginTop": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
          }
        }
      >
        Workouts
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "elevation": 5,
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 4,
        "marginTop": 4,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 18,
          }
        }
      >
        Exercises
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#21897E",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
